### PR TITLE
[ADD] hr_payroll_filter_salary_structure

### PR DESCRIPTION
**Pull request by saltonmassally**
Ability to filter which payroll structure can be selected on contracts

### CODE REVIEW

**Comment by ghost** [comment]
Why don't you simply add a field 'active' on the payroll structure ? The framework will automatically know which structures are selectable. No need for extra domain filter.

**Comment by saltonmassally** [comment]
That would affect structure inheritance. The main point of this was that for complex salary structures you might want some structures to be abstract with the actual structures that can be assigned inheriting from these . An example is the base structure, in most cases you do not want it to assigned to contracts but yet other structures would need to inherit from it

**Comment by ghost** [comment]
@tarzan0820 ha ok

File: hr_payroll_filter_salary_structure/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    "depends": [
+        'hr_payroll'
+    ],
+    "data": [
+        'views/hr_payroll.xml',
+    ],
+    'installable': True,
+}
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (ghost) [inline hr_payroll_filter_salary_structure/__openerp__.py]: remove vim commands please

File: hr_payroll_filter_salary_structure/models/hr_payroll.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+from openerp import models, fields
+
+
+class HrPayrollStructure(models.Model):
+    _inherit = 'hr.payroll.structure'
```
Comment (ghost) [inline hr_payroll_filter_salary_structure/models/hr_payroll.py]: the name of the file should be hr_payroll_structure.py

File: hr_payroll_filter_salary_structure/views/hr_payroll.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
```
Comment (ghost) [inline hr_payroll_filter_salary_structure/views/hr_payroll.xml]: same thing: hr_payroll_structure.xml

**Comment by ghost** [comment]
:+1: code + functional

File: hr_payroll_filter_salary_structure/README.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/hr/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/hr/issues/new?body=module:%20hr_employee_id%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (feketemihai) [inline hr_payroll_filter_salary_structure/README.rst]: Please change the module name in link from hr_employee_id to the name of the module: hr_payroll_filter_salary_structure.

After that :+1:

File: hr_payroll_filter_salary_structure/README.rst [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+Configuration
+=============
+
+This module does not require any special configuration apart from enabling it as
+shown above.
+
+Usage
+=====
+
+
```
Comment (feketemihai) [inline hr_payroll_filter_salary_structure/README.rst:30]: Please add the Try on Runbot section in the Usage paragraph.

File: hr_payroll_filter_salary_structure/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+{
+    "name": "Filter Salary Structure",
+    "version": "0.1",
```
Comment (feketemihai) [inline hr_payroll_filter_salary_structure/__openerp__.py]: Please adjust version number.

**Comment by feketemihai** [comment]
@tarzan0820 @dufresnedavid Do somebody from you continue work on this, do you still need it for v8, there are too many modules on v8, which we don't have a continuation for a long time, so if nobody wants to continue them i think we should close it.

**Comment by feketemihai** [comment]
Please reopen the PR if it's important to you.

**Pull request closed without merging**